modulemd/include/modulemd-2.0/modulemd-eol.h

```diff
@@ -0,0 +1,238 @@
+* For more information on the license, see COPYING.
+* For more information on free software, see <https://www.gnu.org/philosophy/free-sw.en.html>.
+*/
+
+#pragma once
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+/**
+* SECTION: modulemd-eol
+* @title: Modulemd.Eol
+* @stability: stable
+* @short_description: Eol information for a module stream.
+*/
+
+#define MODULEMD_TYPE_EOL (modulemd_eol_get_type ())
+
+G_DECLARE_FINAL_TYPE (ModulemdEol, modulemd_eol, MODULEMD, EOL, GObject)
+
+/**
+ * ModulemdEolVersionEnum:
+ * @MD_EOL_VERSION_ERROR: Represents an error handling mdversion.
+ * @MD_EOL_VERSION_UNSET: Represents an unset mdversion.
+ * @MD_EOL_VERSION_ONE: Represents v1 of the #ModulemdEol metadata
+ * format.
+ * @MD_EOL_VERSION_LATEST: Represents the highest-supported version of the
+ * #ModulemdEol metadata format.
+ *
+ * Since: 2.10
+ */
+typedef enum
+{
+  MD_EOL_VERSION_ERROR = -1,
+
+  MD_EOL_VERSION_UNSET = 0,
+
+  MD_EOL_VERSION_ONE = 1,
+
+  MD_EOL_VERSION_LATEST = MD_EOL_VERSION_ONE,
+} ModulemdEolVersionEnum;
+
+/**
+ * modulemd_eol_new:
+ * @mdversion: The metadata version of this #ModulemdEol.
+ * @modified: The last modified time represented as a 64-bit integer (such as
+ * 201807011200).
+ * @module_name: The name of the module to which this eol applies.
+ * @module_stream: The name of the module stream to which this eol apply.
+ * @message: A string describing the change, reason, etc.
+ *
+ * Returns: (transfer full): A newly-allocated #ModulemdEol object.
+ * This object must be freed with g_object_unref().
+ *
+ * Since: 2.10
+ */
+ModulemdEol *
+modulemd_eol_new (guint64 mdversion,
+                  guint64 modified,
+                  const gchar *module_name,
+                  const gchar *module_stream,
+                  const gchar *message);
+
+/**
+ * modulemd_eol_copy:
+ * @self: This #ModulemdEol object.
+ *
+ * Create a copy of this #ModulemdEol object.
+ *
+ * Returns: (transfer full): The copied #ModulemdEol object.
+ *
+ * Since: 2.10
+ */
+ModulemdEol *
+modulemd_eol_copy (ModulemdEol *self);
+
+/**
+ * modulemd_eol_validate:
+ * @self: This #ModulemdEol object.
+ * @error: (out): If the object is not valid, it will return the reason.
+ *
+ * This method ensures that the eol is internally consistent for usage
+ * or dumping to YAML. It will be run implicitly prior to emitting YAML. This
+ * is not a complete linter, merely a sanity check that the values are not
+ * impossible.
+ *
+ * Since: 2.10
+ */
+gboolean
+modulemd_eol_validate (ModulemdEol *self, GError **error);
+
+/**
+ * modulemd_eol_get_module_context:
+ * @self: (in): This #ModulemdEol object.
+ *
+ * Returns: (transfer none): The context name to which this #ModulemdEol object applies.
+ *
+ * Since: 2.10
+ */
+const gchar *
+modulemd_eol_get_module_context (ModulemdEol *self);
+
+/**
+ * modulemd_eol_set_module_context:
+ * @self: This #ModulemdEol object.
+ * @context: (nullable): The name of the module context to which this eol applies.
```

Comment:
```suggestion
 * @module_context: (nullable): The name of the module context to which this eol applies.
```
Documentation should match actual parameter name.